aspnet/dnx/commands.rst

```diff
@@ -1,8 +1,121 @@
+
+.. note:: The assembly listed in the ``commands`` section should be pulled in by a package that your application depends on.
+
+Installing and creating commands
+--------------------------------
+
+You can add additional commands to your project by modifying the *project.json* file. Additionally, you can use the DNU to install commands globally on a machine using the following syntax::
+
+    dnu commands install MyCommand
+	
+For more details about installing commands, see `DNX Utility <https://github.com/aspnet/Home/wiki/DNX-utility>`_.	
+
+Executing commands
+------------------
+
+To execute commands, you could run them from the command prompt, but a faster way when working with VS Code is to open the **Command Palette** (Ctrl+Shift+P), start typing the name of the command you want to run, and press **Enter**.
+
+You can use DNX to execute the commands defined by your project by entering the following in the command prompt from your project's directory::
+
+    dnx . kestrel
+
+In VS Code, the same command would appear as follows::	
+
+	dnx: kestrel - (SampleWebApp, Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5001
+	
+.. note:: Command execution from the **Command Palette** doesn’t work yet on Linux. To run commands manually on Linux, start a console and type: dnx . <command_name>.
+
+Additionally, you can run a set of standard DNX commands from the command prompt. For example::
+
+	``dnx --appbase c:\myPath\myDirectory``
+
+Building and publishing a command
+---------------------------------
+When you generate a console app using the console app template, it includes a *program.cs* file containing a ``Main`` entry point to the app. After you create a console app, you can build and run the app by issuing the following DNX command::
+
+	dnx . run
+
+In the console app, the *project.json* file contains the ``run`` command in the ``commands`` section. The ``dnx`` command is used to execute a managed entry point (a ``Program.Main`` function) in the assembly. When you issue the above ``dnx . run`` command, DNX finds the command included by the *project.json* file, then finds the ``Main`` entry point that you see in the *program.cs* file. 
+	
+For details about creating a console app with DNX, see :doc:`Creating a Cross-Platform Console App with DNX </dnx/console>`.
+	
+.. note:: The ``dnx . run`` command is a shorthand for executing the entry point in the current project. It is equivalent to ``dnx . [project_name]``. In addition, you can execute any of the commands listed in the *project.json* file from the command window::
+
+	``dnx . (command)``
+	
+To publish commands from your *project.json* file, you can use the ``publish`` command from the DNX Utility. For more details about publishing commands, see `DNX Utility <https://github.com/aspnet/Home/wiki/DNX-utility>`_.
```

Comment:
We should document how to build and pack global commands in this doc. No links to the GitHub wikis allowed.